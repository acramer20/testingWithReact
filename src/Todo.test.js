import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("renders and does not crash", function(){
    render(<Todo/>);
});

it("matches the snapshot", function(){
    const {asFragment} = render(<Todo/>);
    expect(asFragment()).toMatchSnapshot();
});

it("runs the delete function on button click", function() {
    const deleteMock = jest.fn();
    const {getByText} = render(<Todo remove={deleteMock}/>);
    const deleteButton = getByText("X");
    fireEvent.click(deleteButton);
    expect(deleteMock).toHaveBeenCalled();
});