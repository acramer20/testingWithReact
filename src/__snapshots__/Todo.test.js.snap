// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      default todo
    </li>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
